util/qc_executor.py

```diff
@@ -109,6 +138,8 @@ def qc_check(self, parameter, dataset):
 
                 dataset[qc_count_name].values |= flag
                 dataset[qc_results_name].values |= results.astype(np.uint8)
-
-            except (TypeError, ValueError) as e:
-                log.exception('<%s> Failed to execute QC %s %r', self.request_id, function_name, e)
+            # EOFError occurs if qc process dies, therefore failing to write to pipe
+            except EOFError:
+                log.error('<%s> Failed to execute QC %s - qc process died unexpectedly', self.request_id, function_name)
+                if qc_process:
+                    qc_process.join()
```

Comment:
what does calling join here do? if the process died, why join to it again?  This is just for my education, not saying it's wrong, I'm just interested in what's being done here.

Reply:
I don't really understand thread joining as well as I would like, but the thought here was to be conservative and try to join even though the child process should be dead to avoid any edge cases leaving a zombie process.